base/math.jl

```diff
@@ -488,10 +492,12 @@ asinh(x::Number)
 
 # functions that return NaN on non-NaN argument for domain error
 """
-    sin(x)
+    sin(x::T) where T <: Number -> float(T)
```

Comment:
> consistency with the other functions

Which other functions? AFAICT only `getipaddr` and `getipaddrs` have their method signature as shown in the docs written this way:
```
$ sift -e '\bwhere \w+\s*<:\s*\w+\s*->'
stdlib/Sockets/src/addrinfo.jl:246:    getipaddr(addr_type::Type{T}) where T<:IPAddr -> T
stdlib/Sockets/src/addrinfo.jl:280:    getipaddrs(addr_type::Type{T}=IPAddr; loopback::Bool=false) where T<:IPAddr -> Vector{T}
```
My original comment was intended to apply to all additions in this PR rather than just this particular one; apologies, I should have made that clear.

If the `where` is necessary then I think this minor reformatting would be easier would be easier to visually parse as it makes the logical grouping clearer:
```suggestion
    sin(x::T) where {T<:Number} -> float(T)
```

Reply:
There's also `givens` and `isascii`, buit both of those use the curly braces version:

```
[sukera@tower julia]$ rg '\bwhere [\w\s<:]+->'
stdlib/Sockets/src/addrinfo.jl
246:    getipaddr(addr_type::Type{T}) where T<:IPAddr -> T
280:    getipaddrs(addr_type::Type{T}=IPAddr; loopback::Bool=false) where T<:IPAddr -> Vector{T}
[sukera@tower julia]$ rg '\bwhere [\w\s<:{}]+->'
stdlib/Sockets/src/addrinfo.jl
246:    getipaddr(addr_type::Type{T}) where T<:IPAddr -> T
280:    getipaddrs(addr_type::Type{T}=IPAddr; loopback::Bool=false) where T<:IPAddr -> Vector{T}

stdlib/LinearAlgebra/src/givens.jl
270:    givens(f::T, g::T, i1::Integer, i2::Integer) where {T} -> (G::Givens, r::T)

base/strings/basic.jl
632:    isascii(cu::AbstractVector{CU}) where {CU <: Integer} -> Bool
```

I'll add the curly braces, so these are easier to visually distinguish from the intended return type.

Reply:
I was going to make the same comment about the fact that the `where` syntax is quite heavy to read and hides the most useful information. I'd also find it better to use `-> float(typeof(x))`.

Reply:
Removing the `T` entirely will make it confusing where the `T` in `T(NaN)` came from further below in the docstring though.